fml/platform/fuchsia/message_loop_fuchsia.cc

```diff
@@ -32,12 +33,19 @@ void MessageLoopFuchsia::WakeUp(fml::TimePoint time_point) {
     due_time = zx::nsec((time_point - now).ToNanoseconds());
   }
 
-  if (task_.is_pending()) {
-    task_.Cancel();
+  task_mutex_.lock();
```

Comment:
RAII for locking here please. `std::scoped_lock lock(task_mutex_)` would make it unnecessary to manually unlock the thing later.

Reply:
+1